addons/bestja_detailed_reports/models.py

```diff
@@ -0,0 +1,215 @@
+
+
+class ReportEntry(models.Model):
+    _name = 'bestja.report_entry'
+    _order = 'commodity'
+
+    detailed_report = fields.Many2one('bestja.detailed_report', ondelete='cascade', required=True)
+    commodity = fields.Many2one('bestja.commodity_group', string="Nazwa", required=True, ondelete='cascade')
+    commodity_code = fields.Char(string='Kod', related="commodity.code")
+    tonnage = fields.Float(required=True, string="tonaż (kg)")
+    interested_organization = fields.Many2one(
+        'organization',
+        string="Zainteresowana organizacja",
+        store=True,  # Needed by graph view
+        related='detailed_report.interested_organization',
+    )
+    organization = fields.Many2one(
+        'organization',
+        string="Organizacja",
+        store=True,
+        related='detailed_report.project.organization',
+    )
+
+    _sql_constraints = [
+        ('report_entries_uniq',
+        'unique("detailed_report", "commodity")',
+        "Dany element można wybrać tylko raz!")
+    ]
+
+
+class DetailedReport(models.Model):
+    _name = 'bestja.detailed_report'
+    _inherit = [
+        'protected_fields.mixin',
+        'ir.needaction_mixin',
+        'message_template.mixin',
+    ]
+    _order = 'write_uid desc'
+    STATES = [
+        ('sent', "wysłany"),
+        ('accepted', "zaakceptowany"),
+        ('draft', "szkic"),
+        ('rejected', "odrzucony"),
+    ]
+
+    project = fields.Many2one(
+        'bestja.project',
+        required=True,
+        string="Projekt",
+        domain=lambda self: [
+            ('parent.use_detailed_reports', '=', True),
+            ('detailed_reports', '=', False),
+            '|',
+            ('manager', '=', self.env.user.id),
+            ('organization.coordinator', '=', self.env.user.id)],
+        ondelete='restrict',
+    )
+    organization = fields.Many2one(
+        'organization',
+        string="Organizacja",
+        related='project.organization',
+    )
+    name = fields.Char(string="Nazwa projektu", related="project.name")
+    dates = fields.Char(string="Termin", compute="compute_project_dates", store=True)
+    state = fields.Selection(STATES, default='draft', string="Status:")
+    report_entries = fields.One2many('bestja.report_entry', inverse_name='detailed_report', string="Produkt")
+    tonnage = fields.Float(string="Tonaż (kg)", compute="compute_report_tonnage", store=True)
+    parent_project = fields.Many2one(
+        'bestja.project',
+        string="Projekt nadrzędny",
+        related='project.parent',
+        store=True,
+    )
+    interested_organization = fields.Many2one(
+        'organization',
+        string="Zainteresowane organizacje",
+        compute="compute_interested_organization",
+        store=True,
+    )
+    user_can_moderate = fields.Boolean(compute="compute_user_can_moderate")
+
+    @api.one
+    @api.depends('parent_project', 'project')
+    def compute_interested_organization(self):
```

Comment:
ad docstring for the method (it is not easy to deduct what an "interested organization" is)